apps/survey/contracts/Survey.sol

```diff
@@ -98,84 +99,104 @@ contract Survey is AragonApp {
         return _newSurvey(_metadata, _options);
     }
 
+    /**
+     * @notice Reset previously casted vote in survey #`_surveyId`, if any.
+     * @param _surveyId Id for survey
+     */
+    function resetVote(uint256 _surveyId) public {
+        require(canVote(_surveyId, msg.sender));
+
+        SurveyStruct storage survey = surveys[_surveyId];
+        MultiOptionVote storage previousVote = survey.votes[msg.sender];
+        if (previousVote.optionsCastedLength == 0) {
+            return;
```

Comment:
What do you think about using the opposite conditional here, and putting the rest of the logic in that?

E.g.

```
if (previousVote.optionsCastedLength > 0) {
    ... // reset their vote
}
```

I find this structure lets you reason about the functionality a lot easier, especially at a glance. This also lets us avoid conditional comments, like `reset previously voted options, if any`.

Reply:
Well, it was initially this way but I changed it to avoid one unnecessary indentation, which TBH I like it better (and it's a common pattern). I see that condition a little bit like the `require` checks at the beginning of a lot of our functions. So the idea is:
```
function doSomething() {
    // check that you actually can do something or have something to do

    // actually
    // do
    // something
}
```
So when I'm reading the code inside the function I can forget about the conditions, I know everything is fine.
Besides, although it's not much the case here (just one `for`) it helps a lot if the main logic has other conditions or logic structures inside.
But anyway, changing it, I said I would adapt to any style you like ;-)
The only thing I don't get is why should I remove that comment. The function will do exactly the same, so it will only reset previously voted options if any existed (otherwise it will do nothing). I mean, I can remove the comment if yo don't like it, but the same way I could remove it now as the function currently is.

Reply:
For the comment, the `, if any` part makes it sound conditional, but now it isn't :). It'll only reach that block if there were previous votes.

Reply:
I don't get it, so as a general function comment it will apply the same (actually both versions of the function do exactly the same, right?). So *if* there is *any* previous vote, it will reset it, otherwise it won't do anything. 
Maybe it's my English ;-), anyway, removing it, it's just a comment, so it's supposed to be there to help, not to confuse.

Reply:
Oh, damn it! Now I see it! You were talking about the comment *inside* the function, while I was talking about the one in the header! Haha. Sorry for the confusion!